File: remote/src/main/java/com/kelsos/mbrc/presenters/DeviceManagerPresenter.java
package com.kelsos.mbrc.presenters;

import com.kelsos.mbrc.domain.DeviceSettings;
import com.kelsos.mbrc.ui.views.DeviceManagerView;

public interface DeviceManagerPresenter {
  void bind(DeviceManagerView view);

  void onResume();

  void onPause();

  void saveSettings(DeviceSettings settings);

  void loadDevices();

  void deleteSettings(DeviceSettings settings);

  void setDefault(DeviceSettings settings);
}

File: remote/src/main/kotlin/com/kelsos/mbrc/presenters/DeviceManagerPresenter.kt
package com.kelsos.mbrc.presenters

import com.kelsos.mbrc.domain.DeviceSettings
import com.kelsos.mbrc.ui.views.DeviceManagerView

interface DeviceManagerPresenter {
  fun bind(view: DeviceManagerView)

  fun onResume()

  fun onPause()

  fun saveSettings(settings: DeviceSettings)

  fun loadDevices()

  fun deleteSettings(settings: DeviceSettings)

  fun setDefault(settings: DeviceSettings)
}
